src/models/StyledComponent.js

```diff
@@ -298,6 +315,20 @@ export default function createStyledComponent(target: Target, options: Object, r
   // $FlowFixMe
   WrappedStyledComponent.target = isTargetStyledComp ? target.target : target;
 
+  const {
+    style: targetStyleDefaultProp,
+    ...restTargetDefaultProps
+    // $FlowFixMe
+  } = target.targetDefaultProps || {};
+  // $FlowFixMe
+  const { style: styleDefaultProp, ...restDefaultProps } = target.defaultProps || {};
+  // $FlowFixMe
+  WrappedStyledComponent.targetDefaultProps = {
+    ...restTargetDefaultProps,
+    ...restDefaultProps,
+    ...mergeProp('style', targetStyleDefaultProp, styleDefaultProp),
```

Comment:
Can simplify to

```js
const targetDefaultProps = target.defaultProps || EMPTY_OBJECT;

WrappedStyledComponent.targetDefaultProps = {
    ...targetDefaultProps,
    ...mergeProp('style', target.targetDefaultProps.style, targetDefaultProps.style),
};
```

I'm not even sure we need the concept of `targetDefaultProps`. Just set `defaultProps` on the forwardRef right?

Reply:
Hmm yeah, I might be able to optimize I will look at it tomorrow. And I see a case where I might have done something wrong.

```js
const Text = styled.p`
  font-family: ${props => props.font};
`
Text.defaultProps = {
  font: 'sans-serif'
}
```
Here `Text` is `WrappedStyledComponent` so `Text.defaultProps` would actually override it, that's why I used `targetDefaultProps`.